storage/client.go

```diff
@@ -398,6 +398,15 @@ func (c Client) exec(verb, url string, headers map[string]string, body io.Reader
 		return nil, errors.New("azure/storage: error creating request: " + err.Error())
 	}
 
+	// if a body was provided ensure that the content length was set.
```

Comment:
The Content-Length header was set in PutBlockWithLength so it's included when the authentication token is created.